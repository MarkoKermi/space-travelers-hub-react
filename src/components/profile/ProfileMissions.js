import { useSelector } from 'react-redux';
import { Stack } from 'react-bootstrap';
import ProfileMission from './ProfileMission';

const ProfileMissions = () => {
  const missions = useSelector((state) => state.missions.missions);
  const joinedMissions = missions.filter((obj) => obj.isReserved === true);
  return (
    <Stack gap={2}>
      {joinedMissions.map((mission) => (
        <ProfileMission
          key={mission.id}
          id={mission.id}
          name={mission.name}
          isReserved={mission.isReserved}
          wiki={mission.wiki}
        />
      ))}
    </Stack>
  );
};

export default ProfileMissions;
