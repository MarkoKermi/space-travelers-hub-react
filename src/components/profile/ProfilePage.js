import { Container, Row, Col } from 'react-bootstrap';
import ProfileMissions from './ProfileMissions';

const ProfilePage = () => (
  <Container>
    <Row>
      <Col md={6} className="my-5"><ProfileMissions /></Col>
    </Row>
  </Container>
);

export default ProfilePage;
