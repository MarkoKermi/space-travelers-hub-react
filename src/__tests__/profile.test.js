import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import ProfileMission from '../components/profile/ProfileMission';
import ProfileMissions from '../components/profile/ProfileMissions';
import ProfilePage from '../components/profile/ProfilePage';

describe('testing Profile Page components', () => {
  it('testing ProfilePage component', () => {
    const profilePage = renderer.create(<Provider store={store}><ProfilePage /></Provider>);
    expect(profilePage).toMatchSnapshot();
  });

  it('testing ProfileMissions component', () => {
    const profileMissions = renderer.create(<Provider store={store}><ProfileMissions /></Provider>);
    expect(profileMissions).toMatchSnapshot();
  });

  it('testing ProfileMissions component', () => {
    const profileMission = renderer.create(<Provider store={store}><ProfileMission /></Provider>);
    expect(profileMission).toMatchSnapshot();
  });
});