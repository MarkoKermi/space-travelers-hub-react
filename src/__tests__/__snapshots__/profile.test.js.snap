// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Profile Page components testing ProfileMissions component 1`] = `
<div
  className="vstack gap-2"
/>
`;

exports[`testing Profile Page components testing ProfileMissions component 2`] = `
<div
  className="border py-3"
>
  <h3
    className="my-5 fs-3"
  />
  <div
    className="d-flex justify-content-center"
  >
    <button
      className="mx-1 btn btn-outline-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Leave Mission
    </button>
    <a
      className="mx-1 btn btn-outline-primary"
      disabled={false}
      href=""
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
      target="_blank"
    >
      Mission Data
    </a>
  </div>
</div>
`;

exports[`testing Profile Page components testing ProfilePage component 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="my-5 col-sm-6"
    >
      <div
        className="vstack gap-2"
      />
    </div>
  </div>
</div>
`;
